src/main/java/com/example/onlinebookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,82 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.service.BookService;
+import com.example.onlinebookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/categories")
+@Tag(name = "Categories", description = "Operations related to books")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new category",
+            description = "Add a new category to the data")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public CategoryDto createCategory(@RequestBody @Valid CategoryDto categoryDto) {
```

Comment:
don't provide categorydto (response) as a request, create CategoryRequestDto for example

Reply:
comment not fixed